tokio-trace-futures/examples/proxy_server.rs

```diff
@@ -39,11 +39,16 @@ fn main() -> Result<(), Box<std::error::Error>> {
         .map_err(|e| debug!(msg = "error accepting socket", error = field::display(e)))
         .for_each(move |client| {
             let server = TcpStream::connect(&server_addr);
+            let mut client_addr = String::new();
             match client.peer_addr() {
-                Ok(x) => info!(
-                    message = "client connected",
-                    client_addr = field::display(x)
-                ),
+                Ok(x) => {
+                    let x = x.to_string();
```

Comment:
I don't think the `to_string()` is necessary here --- the `fmt::Display` output for `SocketAddr` is [identical to the formatted string](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=acfafb93048d5d623d17745d562486eb) so we could save an allocation by just storing `client_addr` as a `SocketAddr`.

Reply:
Done